numpyro/primitives.py

```diff
@@ -674,7 +698,7 @@ def model(data):
     :rtype: ~jnp.ndarray
     """
     if not _PYRO_STACK:
-        return data
+        return jnp.asarray(data)
```

Comment:
why this is needed? can we just set the return type is ArrayLike?

Reply:
Good point, changed in https://github.com/pyro-ppl/numpyro/pull/1940/commits/b4dacc09b3934c91efc906358c8be6123968d908